### PR TITLE
Woo Stats: Store visitor stats in local cache

### LINKED ISSUE #1310: Woo: Provide option to store visitor stats in local cache
In order to add the scrubbing interaction for the new stats UI changes in woocommerce/woocommerce-android#1252, we need to store the visitor stats in local cache. 

- [x] Create new table and add a new model
- [x] Logic to store the visitor stats for the given time period.
- [x] Logic to retrieve the visitor stats from cache for the given time period
- [x] Add unit tests and mock tests

### PR DESCRIPTION
**Pull request by anitaa1990**
Fixes #1310. This PR adds logic to store the visitor stats response object to the local db in order to support the scrubbing interaction for the charts in the woo app.

### Changes:
- Created a new table and model class - `WCVisitorStatsModel`.
- Added a new Utils class - `WCVisitorStatsSqlUtils` to insert/update/fetch visitor stats from the local db.
- Modified the logic in `OrderStatsApiClient` to store the visitor stats response object to the local db.
- Added method in `WcStatsStore` to retrieve the visitor stats from the local db.
- Added unit tests in `WCVisitorStatsSqlUtilsTest`, `WCStatsStoreTest` and mocked tests in `MockedStack_WCStatsTest`.

### Testing:
- Run all tests in `WCVisitorStatsSqlUtilsTest`, `WCStatsStoreTest` and `MockedStack_WCStatsTest`.
- Please make sure to test upgrading from the version in develop (be sure to pull visitor stats) and then upgrade to this version and pull visitor stats.

### CODE REVIEW

**Comment by anitaa1990** [comment]
Fixed all merged conflicts and PR ready for review 👍

**Comment by AmandaRiu** [comment]
@anitaa1990 looks like there are some conflicts in this branch.

**Comment by anitaa1990** [comment]
Thanks @AmandaRiu! Fixed the merge conflicts and PR ready for review 👍

File: plugins/woocommerce/src/main/kotlin/org/wordpress/android/fluxc/store/WCStatsStore.kt [hunk 1]
```diff
@@ -391,11 +423,16 @@ class WCStatsStore @Inject constructor(
...
 
     private fun handleFetchVisitorStatsCompleted(payload: FetchVisitorStatsResponsePayload) {
-        val granularity = StatsGranularity.fromOrderStatsApiUnit(payload.apiUnit)
-        val onStatsChanged = OnWCStatsChanged(payload.visits, granularity)
-        if (payload.isError) {
-            onStatsChanged.error = payload.error
+        val onStatsChanged = with(payload) {
+            val granularity = StatsGranularity.fromOrderStatsApiUnit(apiUnit)
+            if (isError || stats == null) {
+                return@with OnWCStatsChanged(0, granularity).also { it.error = payload.error }
```
Comment (nbradbury) [inline plugins/woocommerce/src/main/kotlin/org/wordpress/android/fluxc/store/WCStatsStore.kt]: TIL `return@with`.

**Comment by anitaa1990** [comment]
Thank you for the quick review @nbradbury!! 

There were some merge conflicts due to changes in the `develop` branch. I've fixed those and PR is ready for another round @AmandaRiu and @nbradbury  👍

**Comment by nbradbury** [comment]
:shipit:

**Pull request merged**